src/EditorFeatures/Core/RenameTracking/RenameTrackingTaggerProvider.cs

```diff
@@ -86,7 +86,7 @@ internal static bool ResetRenameTrackingStateWorker(Workspace workspace, Documen
                     textBuffer = text.Container.TryGetTextBuffer();
                     if (textBuffer == null)
                     {
-                        FailFast.Fail(string.Format("document with name {0} is open but textBuffer is null. Textcontainer is of type {1}. SourceText is: {2}",
+                        throw new InvalidOperationException(string.Format("document with name {0} is open but textBuffer is null. Textcontainer is of type {1}. SourceText is: {2}",
```

Comment:
Want to turn this into a call to FatalError.Report() or something so we can still have telemetry for this happening in the VS for Windows side of thing?

Reply:
Is the preferred pattern that I throw it here, and then try/catch in the caller with something like `catch (Exception e) when (FatalError.ReportAndPropagateUnlessCanceled(e, cancellationToken))`? That's what I'm seeing elsewhere. Seems a little awkward here given the two one-line callers.

Or should I just do this instead and not propagate the exception at all? Seems low risk to me.

```csharp
var ex = new InvalidOperationException(string.Format("document with name {0} is open but textBuffer is null. Textcontainer is of type {1}. SourceText is: {2}",
    document.Name, text.Container.GetType().FullName, text.ToString()));
FatalError.ReportAndCatch(ex);
return;
```

Reply:
@jasonmalinowski if this is thrown shouldn't it show up as a FatalError.Report anyways with the right stack? I think there will also be a gold bar...

Reply:
@ryzngard Depends: who is catching this?

Reply:
@sandyarmstrong: either is a reasonable option here; the thing I don't really know is if this exception is thrown, who is catching and what is going to happen at that point.

Reply:
Reporting without throwing seems like the way to go here. Will change it.